sdk/src/Core/Amazon.Util/AWSSDKUtils.cs

```diff
@@ -761,14 +762,59 @@ public static long ConvertTimeSpanToMilliseconds(TimeSpan timeSpan)
         /// <returns>String version of the data</returns>
         public static string ToHex(byte[] data, bool lowercase)
         {
-            StringBuilder sb = new StringBuilder();
-
-            for (int i = 0; i < data.Length; i++)
+#if NET8_0_OR_GREATER
+            if (!lowercase)
             {
-                sb.Append(data[i].ToString(lowercase ? "x2" : "X2", CultureInfo.InvariantCulture));
+                return Convert.ToHexString(data);
             }
+#endif
 
-            return sb.ToString();
+#if NETCOREAPP3_1_OR_GREATER
+            Func<int, char> converter = lowercase ? (Func<int, char>)ToLowerHex : (Func<int, char>)ToUpperHex;
+
+            return string.Create(data.Length * 2, (data, converter), (chars, state) =>
```

Comment:
One note: I had to keep this lambda non-static in order to be compatible with `<LangVersion>8</LangVersion>` in the csproj.

Reply:
This should be changed in my opinion. A reasonable version should at least be 9 even when targeting Netstandard 2.0. That is a good language subset without exposing yourself to weird edge cases. That's how also the Azure .NET SDK treated it until they changed it to an even later version as far as I recall.

Reply:
Raised https://github.com/aws/aws-sdk-net/issues/3316